### PR TITLE
Indexes and etags

### PR DESCRIPTION
**Pull request by jwo**
Adds indexes

You can use the `If-Modified-Since` header to check if there 
are any new results since that timestamp. 
If there are no new results, you'll receive a `304 NOT MODIFIED`.

### CODE REVIEW

**Comment by jwo** [comment]
@cmason I seem to recall you did something with etags and stale results (304 NOT MODIFIED)... will you look this over? Things I'm wondering:

* Did I do good/bad with the etag going off of `Shelter.maximum(:updated_at)` rather than the `@shelters.maximum(:updated_at)`  --- I was worried about the filters.
* Do indexes on latitude/longitude help? I _think_ so?

File: db/migrate/20170911205637_add_indexes.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class AddIndexes < ActiveRecord::Migration[5.1]
+  def change
+    add_index :shelters, :active
+    add_index :shelters, :accepting
+    add_index :shelters, :latitude
```
Comment (cmason) [inline db/migrate/20170911205637_add_indexes.rb]: I think the indexes on latitude and longitude should be a composite index.

https://github.com/alexreisner/geocoder#optimisation-of-distance-queries

**Review by cmason — COMMENTED** [review]
I think you want to use `@shelters.maximum(:updated_at)` when checking freshness. I believe the query parameters will constitute a different URI so browsers should treat `/shelters` and `/shelters?limit=42` differently in the browser cache. `Shelter.maximum(:updated_at)` will bust the browser cache whenever a shelter changes. Even if it is  not in the result set.

**Comment by cmason** [comment]
We might want to consider setting the `Cache-Control` header to public so the results can be cached by CDN's and proxy servers.

**Comment by jwo** [comment]
@cmason excellent feedback; believe I made it happen here.

I also found a bug where our `near` and `limit` aren't working nicely together. I investigatied moving to postgis (http://ngauthier.com/2013/08/postgis-and-rails-a-simple-approach.html) but found it wasn't warranted when I could hack together a `take` solution instead.

**Review by cmason — COMMENTED** [review]
Nice catch on the `near` and `limit` bug. I'm going to have to add that into markers too.
👍  🎉

**Pull request merged**